packages/push/src/instruction/instruction_error.rs

```diff
@@ -1,15 +1,26 @@
+use miette::Diagnostic;
+
 use super::IntInstructionError;
 use crate::push_vm::stack::StackError;
 
 /// An error that can occur when performing a `PushInstruction`.
-#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+#[derive(thiserror::Error, Debug, Eq, PartialEq, Diagnostic)]
 pub enum PushInstructionError {
     /// Stack errors can be things like stack over- or underflows.
     #[error(transparent)]
-    StackError(#[from] StackError),
+    StackError(
+        #[from]
+        #[diagnostic_source]
+        StackError,
+    ),
     #[error("Exceeded the maximum step limit {step_limit}")]
+    #[diagnostic(help = "You might want to increase your step limit.")]
```

Comment:
The real issue here is probably that there's an infinite loop, or that evolution has created some _really_ big data structure (like an enormous string) that it's then trying to process.

If you get a `StepLimitExceeded` then it's probably not a "hard" error that would ever make it to the user. Instead it's something that the Push interpreter would process, usually by terminating the execution of this program and (when possible) returning whatever value is on top of the appropriate stack.